packages/pluggableWidgets/datagrid-web/src/Datagrid.xml

```diff
@@ -43,6 +43,19 @@
                                 <caption>Content</caption>
                                 <description/>
                             </property>
+                            <property key="width" type="enumeration" defaultValue="autoFill">
+                                <caption>Column width</caption>
+                                <description />
+                                <enumerationValues>
+                                    <enumerationValue key="autoFill">Auto-fill</enumerationValue>
+                                    <enumerationValue key="autoFit">Auto-fit content</enumerationValue>
+                                    <enumerationValue key="manual">Manual</enumerationValue>
+                                </enumerationValues>
+                            </property>
+                            <property key="size" type="integer" defaultValue="1">
```

Comment:
can't you make it `required`? Then annoying `??` in `Table` can go. And it probably will be ok DX-wise, since you have a default value

Reply:
Not really, size will only be available if manual is selected in the width. So we control over conditional visible properties

Reply:
Yes, but when property is hidden it afaik is reset to default value.

Reply:
I just noticed that is because we are merging Props and PreviewProps, so it can be number | null according to preview types. Better use ! in this case?